### PR TITLE
OS-35: Advance Twig Concepts

### PR DESCRIPTION
**Pull request by shayaanmiyy**
Jira Ticket URL: https://digitalpolygon.atlassian.net/browse/OS-35

# Exercises:

### Twig Macros:

- Create a macro to generate a reusable HTML component (e.g., a button or form).

- Use this macro in multiple templates to ensure consistency and reduce redundancy.

### Custom Filters and Functions:

- Write custom Twig filters and functions to extend its functionality.

- Create a Twig template that uses these custom filters/functions.

### Twig Extensions:

- Explore and integrate a Twig extension (e.g., Twig Extension for additional filters and functions).

- Utilize this extension in a Twig template to enhance its capabilities.

image

Local Screenshot

### CODE REVIEW

File: web/app/themes/dp-timber-theme/index.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
 $context          = Timber::context();
 $context['posts'] = Timber::get_posts();
 $context['foo'] = 'Shayaan';
-$templates        = array( 'index.twig' );
+$templates        = array( 'advance.twig' );
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/index.php]: add a conditional way to provide the new twig suggestion.

File: web/app/themes/dp-timber-theme/views/footer.twig [hunk 2]
```diff
@@ -1 +1,2 @@
-Copyright {{"now"|date('Y')}}
\ No newline at end of file
+Copyright {{"now"|date('Y')}}
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js"></script>
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/views/footer.twig]: Why do we need this now?

File: web/app/themes/dp-timber-theme/views/html-header.twig [hunk 3]
```diff
@@ -12,4 +12,5 @@
     <link rel="author" href="{{ site.theme.link }}/humans.txt" />
     <link rel="pingback" href="{{ site.pingback_url }}" />
     <link rel="profile" href="http://gmpg.org/xfn/11">
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/views/html-header.twig]: Do we need this anymore as we are adding this in different way.

File: web/app/themes/dp-timber-theme/views/footer.twig (hunk 2 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/views/footer.twig]: No we don't need this, I taught having an simpler alternative way, I will remove it

File: web/app/themes/dp-timber-theme/views/html-header.twig (hunk 3 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/views/html-header.twig]: Yes, I will remove it

File: web/app/themes/dp-timber-theme/index.php (hunk 1 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/index.php]: I will go with the partials & Include method as we have done it in basic syntax

**Comment by Prashant114606** [comment]
@shayaanmiyy could you please resolve the conflicts ?

**Pull request merged**